include/unifex/sync_wait.hpp

```diff
@@ -125,27 +105,18 @@ using receiver_t = typename _receiver<T>::type;
 
 template <typename Result, typename Sender>
 std::optional<Result> _impl(Sender&& sender) {
-  auto blockingResult = blocking(sender);
-  const bool completesSynchronously =
-      blockingResult == blocking_kind::always ||
-      blockingResult == blocking_kind::always_inline;
-
   using promise_t = _sync_wait::promise<Result>;
   promise_t promise;
-  if (!completesSynchronously) {
-    promise.doneEvent_.emplace();
-  }
+  manual_event_loop ctx;
 
   // Store state for the operation on the stack.
   auto operation = connect(
-      ((Sender &&) sender),
-      _sync_wait::receiver_t<Result>{promise});
+      with_query_value((Sender&&)sender, get_scheduler, ctx.get_scheduler()),
```

Comment:
Indeed, although this seemed marginally easier to read as a way to get the customisation in there.

I don't really understand why this approach would cause tag_invoke to be ambiguous, though

Reply:
As it turns out, that does fix the ambiguity... hmm.